python/ml4ir/base/data/ranklib_reader.py

```diff
@@ -0,0 +1,95 @@
+import os
+import pandas as pd
+import tensorflow as tf
+
+from ml4ir.base.io.file_io import FileIO
+from ml4ir.base.features.feature_config import FeatureConfig
+from ml4ir.base.data import tfrecord_reader, tfrecord_writer, ranklib_to_ml4ir
+
+from typing import List
+
+TFRECORD_FILE = "file_0.tfrecord"
+
+def read(
+    data_dir: str,
+    feature_config: FeatureConfig,
+    tfrecord_type: str,
+    tfrecord_dir: str,
+    file_io: FileIO,
+    batch_size: int = 128,
+    preprocessing_keys_to_fns: dict = {},
+    use_part_files: bool = False,
+    max_sequence_size: int = 25,
+    parse_tfrecord: bool = True,
+    logger=None,
+    keep_additional_info = 0,
+    non_zero_features_only = 1,
+    **kwargs
+) -> tf.data.TFRecordDataset:
+    """
+    - reads ranklib-formatted data from an input directory
+    - selects relevant features
+    - creates Dataset X and y
+
+    Current execution plan:
+        1. Conver ranklib to a dataframe
+        2. Convert each query into tf.train.SequenceExample protobufs
+        3. Write the protobufs into a .tfrecord file
+        4. Load .tfrecord file into a TFRecordDataset and parse the protobufs
+
+    Args:
+        - data_dir: Path to directory containing csv files to read
+        - feature_config: ml4ir.config.features.FeatureConfig object extracted from the feature config
+        - tfrecord_dir: Path to directory where the serialized .tfrecord files will be stored
+        - batch_size: int value specifying the size of the batch
+        - use_part_files: bool value specifying whether to look for part files
+        - max_sequence_size: int value specifying max number of records per query
+        - logger: logging object
+        - keep_additional_info: Option to keep additional info (All info after the "#") 1 to keep, 0 to ignore
+        - non_zero_features_only: Only non zero features are stored. 1 for yes, 0 otherwise
```

Comment:
@mohazahran Can this not be done from the feature config YAML?

Reply:
These are passed as command line arguments